test/puppetlabs/puppetdb/testutils/services.clj

```diff
@@ -352,3 +355,28 @@
   (until-consumed
    (fn []
      (pdb-client/submit-command-via-http! base-url cmd version payload))))
+
+(defn wait-for-server-processing
+  "Returns a truthy value indicating whether the wait was
+  successful (i.e. didn't time out).  Given a false or nil timeout,
+  waits forever."
+  ([server] (wait-for-server-processing server nil))
+  ([server timeout-ms]
+   (let [now-ms #(time-coerce/to-long (t/now))
+         deadline (and timeout-ms (+ (now-ms) timeout-ms))]
+     (loop []
+       (cond
+         (and deadline (> (now-ms) deadline))
+         false
+
+         (let [srv (get-service server :PuppetDBCommandDispatcher)
+               stats (dispatch/stats srv)]
+           (= (:executed-commands stats) (:received-commands stats)))
+         true
+
+         :else ;; In theory, polling might be avoided via watcher.
+         (do
+           (Thread/sleep (if-not deadline
+                           100
+                           (max 0 (min 100 (- deadline (now-ms))))))
```

Comment:
And I forgot, @mullr asked for this function to be busier in its previous incarnation, so I'll just change it to 10ms (as I did there), and omit the extra work.